src/Core/Form/IdentifiableObject/DataProvider/CategoryFormDataProvider.php

```diff
@@ -101,15 +103,11 @@ public function getData($categoryId)
      */
     public function getDefaultData()
     {
-        $defaultGroups = $this->defaultGroupsProvider->getGroups();
+        $allGroupIds = array_column(Group::getGroups((int) Configuration::get('PS_LANG_DEFAULT')), 'id_group');
```

Comment:
Please could you add the configuration adapter to the constructor for avoiding legacy classes ?

Reply:
@Progi1984 I've refacto my pr to another approch : I use defaultGroupsProvider to get default groups pour this case : all ids